src/collector/parsers/typescript.ts

```diff
@@ -201,19 +201,17 @@ function createFunction(node: ts.MethodDeclaration | ts.FunctionDeclaration): IF
 function createMethod(isInternal = false) {
     return function (node: ts.MethodDeclaration): IMethod {
         const isPublicMethod = VisibilityMap[ts.SyntaxKind.PublicKeyword] === extractVisibility(node);
+        const methodOutput = {
+            ...createFunction(node),
+            visibility: extractVisibility(node)
+        };
 
         if (!isInternal && isPublicMethod) {
-            return {
-                ...createFunction(node),
-                visibility: extractVisibility(node)
-            }
+            return methodOutput;
         }
 
         if (isInternal && !isPublicMethod) {
-            return {
-                ...createFunction(node),
-                visibility: extractVisibility(node)
-            }
+            return methodOutput;
         }
```

Comment:
It will be better to use one if block:
if ((!isInternal && isPublicMethod) || (isInternal && !isPublicMethod)) {
    return {
                ...createFunction(node),
                visibility: extractVisibility(node)
            }
            return methodOutput;
}

Reply:
Suggest to use basic XOR operator to simplify this condition:
```
if (Boolean(isInternal ^ isPublicMethod)) {
```

or you can even just compare two values:
```
if (isInternal !== isPublicMethod) {
```

But in any case I suggest to move condition into constant to describe what exactly we are checking.